Adds properties on edges

andrejtonev opened the pull request:
Define relationship properties in many_to_many via name_mappings and properties field

antepusic reviewed (changes requested):
The code looks fine. Can you add a test to serve as an example and verify that this feature works?

andrejtonev commented:
@antepusic I changed the code a bit. Had some CI problems, not sure why it didn't like the first version...

katarinasupe commented:
Hi @andrejtonev, the related docs need to be properly updated, and you should rebase to main. We no longer use the develop branch for the release; everything goes directly to the main.


FYI, as discussed, this will be included in the next release (July 3rd). Work on Memgraph is a priority, so we can move the milestone due date if you can't make it.

andrejtonev commented:
one_to_many: passing field parameters (which already existed but was not used) many_to_many: changed from parameters to column_names_mapping
	      using the mapping and columns to define properties
	      ignoring from to keys when adding properties

### Description

Responding to user-made issue. Requesting the ability to define properties on edges.
Something similar was already present, but was not used and needed to be expanded to support mappings.
Mappings are only supported for many_to_many
one_to_many only supports predefined and fixed parameters defined via the configuration file.

### Pull request type

Please delete options that are not relevant.

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring with functional or API changes
- [ ] Refactoring without functional or API changes
- [ ] Build or packaging related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Related issues

Delete section if this PR doesn't resolve any issues. 

Closes #301 

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

######################################

### Reviewer checklist (the reviewer checks this part)
- [ ] Core feature implementation
- [ ] Tests
- [ ] Code documentation
- [ ] Documentation on [gqlalchemy/docs](https://github.com/memgraph/gqlalchemy/tree/main/docs)

######################################

andrejtonev commented:
Do not merge
Need to update implementation.
1. Use name mappings instead of a local column name map.
2. ~Figure out how/if to implement one_to_many edge parameters~ Remove one_to_many edge parameters. We cannot add information not present in one of the nodes (edge created via node creation triggers). We could add a fixed param, but I just don't see a good use-case at the moment.

andrejtonev commented:
@antepusic @Josipmrden I didn't like my previous implementation. Made a few changes. Please take another look
Thanks

antepusic reviewed (approved):
The changes make sense, well done

andrejtonev commented:
@katarinasupe I updated the relevant docs.
Changed the example for the one_to_many and added one for many_to_many.

katarinasupe commented on docs/reference/gqlalchemy/transformations/importing/loaders.md:
```diff
@@ -44,7 +43,7 @@ Many to many mapping is intended to be used in case of associative tables.
 - `foreign_key_to` - Describes the destination of the relationship.
 - `label` - Label to be applied to the newly created relationship.
-- `parameters` - Parameters that will be added to the relationship created from this object (Optional).
+- `parameters` - List of parameters that will be added to the relationship created from this object (Optional).
```
In the code, you referred to this as Properties instead of a List of parameters. Which one is the correct one? Maybe List of properties?

andrejtonev replied:
Until now, the `parameters` variable was actually just ignored.
So let's just "break" the api and rename this to `properties`.
Agree?

katarinasupe replied:
I vote 'yes' 😄

katarinasupe commented on gqlalchemy/transformations/importing/loaders.py:
```diff
@@ -616,6 +618,8 @@ def _save_row_as_relationship(
             property_to: Property of the destination node.
             relation_label: Label for the relationship.
+            table_name: Name of the table used to read parameters
+            parameters: Relationship parameters to be added
```
Again here, a bit of a different definition. It makes more sense to say "Relationship properties to be added". How did you decide on the `parameters` to begin with?

katarinasupe commented:
@andrejtonev Merge the latest main for tests to pass. 
Once the tests pass, feel free to squash and merge.
